subgraph/src/public-lock.ts

```diff
@@ -314,3 +323,28 @@ export function handleLockMetadata(event: LockMetadataEvent): void {
     lock.save()
   }
 }
+/**
+ * Create Receipt object for subgraph for key 'purchase'/'extend'/'renewal'
+ * @param {String} keyID - key id
+ * @param {event} Object - Object event
+ * @return {void}
+ */
+export function handleKeyReceipt(
+  keyID: string | number,
+  event: RenewKeyPurchaseEvent | TransferEvent | KeyExtendedEvent
+): void {
+  const receipt = Receipt.load(`${keyID}`)
+  const lock = Lock.load(event.address.toHexString())
+
+  if (receipt) {
+    receipt.hash = event.transaction.hash.toString()
+    receipt.timestamp = event.block.timestamp
+    receipt.payer = event.transaction.from.toString() // address who pays for the membership
+    receipt.sender = event.transaction.to?.toString() || null // address will sends the transaction
```

Comment:
The payer is the address who spends the ERC20 and it is can be different from the address of the recipient of the NFT